src/windows/BarcodeScannerProxy.js

```diff
@@ -570,7 +580,25 @@ module.exports = {
             }
         }
 
-        BarcodeReader.scanPromise = WinJS.Promise.wrap(createPreview())
+        function errorHandler(error) {
+            // Suppress null result (cancel) on suspending
+            if (BarcodeReader.suspended) {
+                return;
+            }
+
+            destroyPreview();
+            if (error.message == 'Canceled') {
+                success({
+                    cancelled: true
+                });
+            } else {
+                fail(error);
+            }
+        }
+
+        // Timeout is needed so that the .done finalizer below has some time 
+        // to be attached to the promise.
+        BarcodeReader.scanPromise = WinJS.Promise.timeout(200, WinJS.Promise.wrap(createPreview()))
```

Comment:
Not sure, why timeout is required, since composing a promise chain is sync, but perhaps this has something to do with sync `createPreview` being wrapped into promise. You might want to make `createPreview` return promise instead of this workaround.